lib/ruby-next/language/rewriters/squiggly_heredoc.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module RubyNext
+  module Language
+    module Rewriters
+      class SquigglyHeredoc < Base
+        NAME = "squiggly-heredoc"
+        SYNTAX_PROBE = "txt = <<~TXT\n  bla\n      TXT"
+        MIN_SUPPORTED_VERSION = Gem::Version.new("2.3.0")
+
+        def on_str(node)
+          return node unless node.loc.respond_to?(:heredoc_body) && node.loc.expression.source.include?("<<~")
+
+          context.track! self
+
+          replace(node.loc.expression, node.loc.expression.source.tr("~", "-"))
+
+          node
+        end
+
+        alias on_dstr on_str
```

Comment:
same here, `dstr` has interpolation, and so nested squiggly heredocs are ignored. same with `on_xstr`

Reply:
> so nested squiggly heredocs are ignored

Ha, nested heredocs 🙃 I don't want to invest a lot of time into very old Rubies support, so nested heredoc users are going to continue suffer 😈)

Also, I found that there is no `on_str` method but we do have `str` nodes. Is it a bug in Parser or is it intentional?

Reply:
> Also, I found that there is no on_str method but we do have str nodes. Is it a bug in Parser or is it intentional?

Yeah, I was thinking about it a few times. This node is a leaf, and so a visitor should be empty. 

By adding an empty method we will slightly slow down `Parser::AST::Processor` because of this line - https://github.com/whitequark/ast/blob/master/lib/ast/processor/mixin.rb#L258 (before looking down into the code I thought there's no `handler_missing`, but anyway, static dispatch is faster than dynamic because of inline method caches, JIT optimizations etc)

On the other side, users have to remember which nodes are leaves to not accidentally call `super` in the custom handler (same for `int`, `float` and other trivial nodes).

Both pros and cons are way to negligible to be real, so for me the only argument is my laziness. Feel free to send a PR to parser if you need them.

Also, does `super` in the `alias`-ed method call the same method on parent? I mean, does this code
```
class Parser::AST::Processor
  def on_str(node); end
end

class CustomVisitor
  def capture_something(node)
    # generic code that calls `super`
  end

  alias on_str capture_something
end
```

call `Parser::AST::Processor#on_str`? I'm not 100% sure, but IIRC `alias` reuses original method.

Reply:
> call Parser::AST::Processor#on_str? 

Nope, it will try to call `#capture_something`.

> Feel free to send a PR to parser if you need them.

I can use `defined?(super_method)` for now )